jaxrs-tck/src/main/java/jakarta/ws/rs/tck/jaxrs21/ee/client/rxinvoker/JAXRSClientIT.java

```diff
@@ -0,0 +1,1975 @@
+   * 
+   * @assertion_ids: JAXRS:JAVADOC:1155; JAXRS:JAVADOC:1162; JAXRS:JAVADOC:1185;
+   * 
+   * @test_Strategy: ResponseProcessingException - in case processing of a
+   * received HTTP response fails (e.g. in a filter or during conversion of the
+   * response entity data to an instance of a particular Java type).
+   */
+  @Test
+  public void putWithGenericTypeThrowsProcessingExceptionTest() throws Fault {
+    _hostname = NONEXISTING_SITE;
+    GenericType<String> generic = createGeneric(String.class);
+    Entity<String> entity = Entity.entity("put", MediaType.WILDCARD_TYPE);
+    CompletionStageRxInvoker rx = startRxInvokerForMethod("put");
+    Future<String> future = rx.put(entity, generic).toCompletableFuture();
+    assertExceptionWithProcessingExceptionIsThrownAndLog(future);
+  }
+
+  /*
+   * @testName: putWithGenericTypeThrowsWebApplicationExceptionTest
+   * 
+   * @assertion_ids: JAXRS:JAVADOC:1155; JAXRS:JAVADOC:1162; JAXRS:JAVADOC:1185;
+   * 
+   * @test_Strategy: WebApplicationException - in case the response status code
+   * of the response returned by the server is not successful and the specified
+   * response type is not Response.
+   */
+  @Test
+  public void putWithGenericTypeThrowsWebApplicationExceptionTest()
+      throws Fault {
+    GenericType<String> generic = createGeneric(String.class);
+    Entity<String> entity = Entity.entity("put", MediaType.WILDCARD_TYPE);
+    CompletionStageRxInvoker rx = startRxInvokerForMethod("putnotok");
+    Future<String> future = rx.put(entity, generic).toCompletableFuture();
+    assertExceptionWithWebApplicationExceptionIsThrownAndLog(future);
+  }
+
+  /*
+   * @testName: putWithGenericTypeThrowsNoWebApplicationExceptionForResponseTest
+   * 
+   * @assertion_ids: JAXRS:JAVADOC:1155; JAXRS:JAVADOC:1162; JAXRS:JAVADOC:1185;
+   * 
+   * @test_Strategy: WebApplicationException - in case the response status code
+   * of the response returned by the server is not successful and the specified
+   * response type is not Response.
+   */
+  @Test
+  public void putWithGenericTypeThrowsNoWebApplicationExceptionForResponseTest()
+      throws Fault {
+    Entity<String> entity = Entity.entity("put", MediaType.WILDCARD_TYPE);
+    CompletionStageRxInvoker rx = startRxInvokerForMethod("putnotok");
+    GenericType<Response> generic = createGeneric(Response.class);
+    Future<Response> future = rx.put(entity, generic).toCompletableFuture();
+    checkFutureResponseStatus(future, Status.NOT_ACCEPTABLE);
+  }
+
+  // ------------------------------------------------------------------
+  // ---------------------------TRACE -----------------------------------
+  // ------------------------------------------------------------------
+
+  /*
+   * @testName: traceTest
+   * 
+   * @assertion_ids: JAXRS:JAVADOC:1156; JAXRS:JAVADOC:1162; JAXRS:JAVADOC:1186;
+   * 
+   * @test_Strategy: Invoke HTTP trace method for the current request
+   */
+  //@Test
```

Comment:
Same here, should it be excluded?

Reply:
I have tracked this failure under #1020 for now.

Reply:
I believe that it is imperative not to comment out the TRACE tests. No matter whether they fail. They fail because in GF HTTP trace is disabled by default. It can be enabled by 
```
            <plugin>
                <artifactId>exec-maven-plugin</artifactId>
                <groupId>org.codehaus.mojo</groupId>
                <executions>
                    <execution>
                        <id>StartDomain</id>
                        <phase>pre-integration-test</phase>
                        <goals>
                            <goal>exec</goal>
                        </goals>
                        <configuration>
                            <workingDirectory>${project.build.directory}/glassfish6/glassfish/bin</workingDirectory>
                            <executable>asadmin</executable>
                            <arguments>
                                <argument>start-domain</argument>
                            </arguments>
                        </configuration>
                    </execution>
                    <execution>
                        <id>Enable Trace requests</id>
                        <phase>pre-integration-test</phase>
                        <goals>
                            <goal>exec</goal>
                        </goals>
                        <configuration>
                            <workingDirectory>${project.build.directory}/glassfish6/glassfish/bin</workingDirectory>
                            <executable>asadmin</executable>
                            <arguments>
                                <argument>set</argument>
                                <argument>server-config.network-config.protocols.protocol.http-listener-1.http.trace-enabled=true</argument>
                            </arguments>
                        </configuration>
                    </execution>
                    <execution>
                        <id>StopDomain</id>
                        <phase>pre-integration-test</phase>
                        <goals>
                            <goal>exec</goal>
                        </goals>
                        <configuration>
                            <workingDirectory>${project.build.directory}/glassfish6/glassfish/bin</workingDirectory>
                            <executable>asadmin</executable>
                            <arguments>
                                <argument>stop-domain</argument>
                            </arguments>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

Reply:
But other vendors do run the tests and do not have the trace disabled by default. It is very easy to forget to enable the tests again.

Reply:
Thanks. I enabled the GF http trace using above and the trace tests do pass.